Runtime/Data/Type/Transformation/Vector2MagnitudeSetter.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Zinnia.Data.Type.Transformation
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using Malimbe.PropertySerializationAttribute;
+    using Malimbe.XmlDocumentationAttribute;
+
+    /// <summary>
+    /// Transforms a <see cref="Vector2"/> by changing its magnitude.
+    /// </summary>
+    public class Vector2MagnitudeSetter : Transformer<Vector2, Vector2, Vector2MagnitudeSetter.UnityEvent>
```

Comment:
it is a bit vague of what qualifies as a Transformation (semantically).
let's assume it will only be Conversion and Aggregation.

there is another namespace `Operation`. it has `Mutation` (to change a property/field value on something) and `Extraction`.
The `Extraction` is again quite vague as of now. for example, it has GameObjectExtractors and TransformVector3PropertyExtractors. Along side, there is a `Vector2ComponentExtractor` which is actually doing the same job to `Vector2ToFloat`.

I think some of them are harder to be grouped up
`VectorXMagnitudeSetter`, `VectorXToFloat(mag,sqrMag)`, `FloatRangeValueRemapper`
They seem to be just a kind of `Operation`.

Reply:
* Transformation - Take existing data type values and create a new value
  * Conversions convert from one type to another (e.g. bool to float).
  * Aggregations use a collection of the same data type and perform an aggregate function on them, which is to take a collection of data and create a single summary value.
* Operation - Take an existing data type value and do something with it without creating a new data type or result
  * Mutators take an existing data type and mutate the data on that data type
  * Extractors pull out a piece of data from a complex data type.

> I agree things like the `Vector2ComponentExtractor` is most likely a duplicate of the `Vector2ToFloat` and should probably be looked into retiring the extractor.

The existing `VectorXMagnitudeSetter` doesn't seem to fall into any of the above 4 categories, but the issue isn't that it doesn't fall into the category, it's the name isn't describing what it is.

It's not setting the magnitude on a VectorX - because if it was then it would be mutating the VectorX and therefore would fall under Mutator.

It's also not converting a VectorX into a different data type.

What it is doing is taking an existing VectorX and creating a new version of that VectorX with modifications to its data. So it's not a clone action either.

So fundamentally the issue is the word `Setter` because it's not a setter. It's fine if there is no existing namespace outside of Operation for it, but it needs to be defined as to what it actually is doing.

Reply:
If this was to be a `Vector2Cloner`

then you could potentially have a class that has an `Override` (better name maybe?) property for each property of the Vector2:

```
Magnitude.Override
Normalized.Override
SqrMagnitude.Override
X.Override
Y.Override
```

Then you know what's going to happen is you call `Process` and it emits the cloned Vector2, but you can override any of the properties in that Vector2 prior to the cloning happening.

Reply:
What is the actual purpose of the class though? Why do you want to clone a Vector2 and change the magnitude of it?

Because if a magnitude is the length of the vector and the X and Y are lets say `1,1`

then they will have a specific magnitude.

But if you now say actually the magnitude is something not of the product of `1,1` then isn't your Vector2 now just wrong?

What am I missing?